### PR TITLE
[VEN-1996]: add agEUR market in Stablecoins pool simulations

### PR DESCRIPTION
**Pull request by GitGuru7**
Resolves VEN-1996

### CODE REVIEW

File: vips/vip-178/vip-178.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+const agEUR = "0x12f31b73d812c6bb0d735a218c086d44d5fe5f89";
+const POOL_REGISTRY = "0x9F7b01A536aFA00EF10310A162877fd792cD0666";
+const VTOKEN_RECEIVER_agEUR = "0xc444949e0054a23c44fc45789738bdf64aed2391";
+const vagEUR_StableCoin = "0x1a9D2862028F6f5E6C299A7AC3C285508942b15E";
+const BINANCE_ORACLE = "0x594810b741d136f1960141C0d8Fb4a91bE78A820";
+const RESILIENT_ORACLE = "0x6592b5DE802159F3E74B2486b091D11a8256ab8A";
+const TREASURY = "0xF322942f644A996A617BD29c16bd7d231d9F35E9";
+const NORMAL_TIMELOCK = "0x939bD8d64c0A9583A7Dcea9933f7b21697ab6396";
+const MAX_STALE_PERIOD = 60 * 25;
```
Comment (chechu) [inline vips/vip-178/vip-178.ts:14]: The heartbeat of `agEUR` is 60 minutes according to Binance Oracle: https://oracle.binance.com/data-feeds/detail/AGEUR-USD

So, I would set a max stale period of 100 minutes in this case, following the ratio 1.6 we are following when the heartbeat is 15 minutes.

Comment (GitGuru7) [reply vips/vip-178/vip-178.ts]: [b15b6a0](https://github.com/VenusProtocol/vips/pull/79/commits/b15b6a0093c4bae9aeeb0f16eefeda428d9b0198)

File: vips/vip-178/vip-178.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            VTOKEN_RECEIVER_agEUR,
+            parseUnits("100000", 18),
+            parseUnits("50000", 18),
+          ],
+        ],
+      },
+      {
+        target: TREASURY,
+        signature: "withdrawTreasuryBEP20(address,uint256,address)",
+        params: [agEUR, parseUnits("17650", 18), NORMAL_TIMELOCK],
```
Comment (chechu) [inline vips/vip-178/vip-178.ts:111]:
```suggestion
        params: [ANGLE, parseUnits("17650", 18), NORMAL_TIMELOCK],
```

File: vips/vip-178/vip-178.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      },
+      {
+        target: agEUR,
+        signature: "approve(address,uint256)",
+        params: [POOL_REGISTRY, "0"],
+      },
+      {
+        target: agEUR,
+        signature: "approve(address,uint256)",
+        params: [POOL_REGISTRY, parseUnits("10000", 18)],
```
Comment (chechu) [inline vips/vip-178/vip-178.ts:91]:
```suggestion
        params: [POOL_REGISTRY, parseUnits("9000", 18)],
```

File: vips/vip-178/vip-178.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            ],
+            [true, false, false],
+          ],
+        ],
+      },
+
+      {
+        target: BINANCE_ORACLE,
+        signature: "setMaxStalePeriod(string,uint256)",
+        params: ["ANGLE", maxStalePeriod || MAX_STALE_PERIOD],
```
Comment (chechu) [inline vips/vip-178/vip-178.ts:60]: The heartbeat of `ANGLE` is 15 minutes, different from the heartbeat of `agEUR` (60 minutes). So, I would set different max stale periods for each one

Comment (GitGuru7) [reply vips/vip-178/vip-178.ts]: [ab37758](https://github.com/VenusProtocol/vips/pull/79/commits/ab3775867d5e5c64694a0003f69ca486347f62b4)

File: vips/vip-178/vip-178.ts (hunk 3 above)
Comment (GitGuru7) [reply vips/vip-178/vip-178.ts]: [ab37758](https://github.com/VenusProtocol/vips/pull/79/commits/ab3775867d5e5c64694a0003f69ca486347f62b4)

File: vips/vip-178/vip-178.ts (hunk 2 above)
Comment (GitGuru7) [reply vips/vip-178/vip-178.ts]: [ab37758](https://github.com/VenusProtocol/vips/pull/79/commits/ab3775867d5e5c64694a0003f69ca486347f62b4)

File: vips/vip-178/vip-178-testnet.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        signature: "setDirectPrice(address,uint256)",
+        params: [agEUR, parseUnits("1.06", 18)],
+      },
+      {
+        target: CHAINLINK_ORACLE,
+        signature: "setDirectPrice(address,uint256)",
+        params: [ANGLE, parseUnits(".032", 18)],
+      },
+      {
+        target: RESILIENT_ORACLE,
```
Comment (chechu) [inline vips/vip-178/vip-178-testnet.ts:39]: I miss the `setTokenConfig` command for the `ANGLE` token

File: simulations/vip-178/vip-178/simulations.ts [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+    });
+
+    describe("Initial supply", () => {
+      it(`should mint 9,000 vagEUR to ${VTOKEN_RECEIVER_agEUR}`, async () => {
+        expect(await vagEUR.balanceOf(VTOKEN_RECEIVER_agEUR)).to.equal(parseUnits("9000", 8));
+      });
+    });
+
+    describe("Market and risk parameters", () => {
+      const checkInterestRate = (
```
Comment (chechu) [inline simulations/vip-178/vip-178/simulations.ts:148]: We could move this check function to this folder, to reuse it in the future: https://github.com/VenusProtocol/vips/blob/develop/src/vip-framework/checks

File: simulations/vip-178/vip-178/simulations.ts [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+
+  before(async () => {
+    poolRegistry = await ethers.getContractAt(POOL_REGISTRY_ABI, POOL_REGISTRY);
+    comptroller = await ethers.getContractAt(COMPTROLLER_ABI, STABLECOIN_COMPTROLLER);
+    vagEUR = await ethers.getContractAt(VTOKEN_ABI, VagEUR_Stablecoins);
+    rewardsDistributor = await ethers.getContractAt(REWARD_DISTRIBUTOR_ABI, REWARD_DISTRIBUTOR);
+  });
+
+  describe("Contracts setup", () => {
+    const checkVToken = (
```
Comment (chechu) [inline simulations/vip-178/vip-178/simulations.ts:44]: We could move this check function to this folder, to reuse it in the future: https://github.com/VenusProtocol/vips/blob/develop/src/vip-framework/checks

File: simulations/vip-178/vip-178/simulations.ts [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+          });
+
+          it("should set VagEUR_Stablecoins supply cap to 1,00,000", async () => {
+            expect(await comptroller.supplyCaps(VagEUR_Stablecoins)).to.equal(parseUnits("100000", 18));
+          });
+        });
+      });
+    });
+
+    describe("Reward Distributor", () => {
```
Comment (chechu) [inline simulations/vip-178/vip-178/simulations.ts:234]: We could use the check that we have here: https://github.com/VenusProtocol/vips/blob/develop/src/vip-framework/checks/rewardsDistributor.ts

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/178

**Review by chechu — APPROVED** [review]

**Pull request merged**